syntax = "proto3";

option cc_generic_services = true;

import "google/protobuf/empty.proto";
import "service/DataService.proto";
import "service/GachaService.proto";
package apb.api.gameplay;

option java_package = "jp.co.apb.server.api.proto.gameplay";
option java_multiple_files = true;
option csharp_namespace = "Art.Framework.ApiNetwork.Grpc.Api.GamePlay";

message CheckBeforeGamePlayRequest {
  string tr = 1;
  int32 voice_client_system_language_type_id = 2;
  int32 text_client_system_language_type_id = 3;
}

message CheckBeforeGamePlayResponse {
  bool is_exist_unread_pop = 1;
  repeated .apb.api.gacha.MenuGachaBadgeInfo menu_gacha_badge_info = 2;
  map<string, .apb.api.data.DiffData> diff_user_data = 99;
}

service GamePlayService {
  rpc CheckBeforeGamePlay(.apb.api.gameplay.CheckBeforeGamePlayRequest) returns (.apb.api.gameplay.CheckBeforeGamePlayResponse);
}

